IdentityCore/src/cache/mac/MSIDMacKeychainTokenCache.m

```diff
@@ -516,30 +450,71 @@ - (MSIDCredentialCacheItem *)tokenWithKey:(__unused MSIDCacheKey *)key
                                               context:(__unused id<MSIDRequestContext>)context
                                                 error:(__unused NSError **)error
 {
-    [self createUnimplementedError:error context:context];
-    return nil;
+    NSArray *tokenItems = [self itemsWithKey:key context:context error:error];
+
+    if (!tokenItems)
+    {
+        return nil;
+    }
+
+    NSMutableArray<MSIDCredentialCacheItem *> *tokenList = [NSMutableArray new];
+    for (NSDictionary *dict in tokenItems)
+    {
+        NSData *jsonData = dict[(id)kSecValueData];
+        if (jsonData)
+        {
+            MSIDCredentialCacheItem *cacheItem = (MSIDCredentialCacheItem *)[serializer deserializeCredentialCacheItem:jsonData];
```

Comment:
nit: this seems like a common code across iOS and Mac. Do we want to have a base class for both keychain caches to handle this and other common logic?

Reply:
I agree. But I would prefer it to be a separate PR. It would include many other changes. Lets discuss offline